scripts/classification/imagenet/train_imagenet.py

```diff
@@ -13,18 +12,26 @@
 from gluoncv.model_zoo import get_model
 from gluoncv.utils import makedirs, LRSequential, LRScheduler
 
+import dali
```

Comment:
can you ensure that ImportError of dali will be catched properly? 
Not everyone will use the bash file as entry point. Basically we should show a warning or message to suggest that user can install dali via pip if they want the maximized training speed.

Reply:
@zhreshold : 
- I moved the installation of DALI from bash file into `train_imagenet.py`. So, when `--data-backend` is equal to `dali-gpu` or `dali-cpu` AND DALI is not installed, it will be installed from Python part.
- I alse added check for `--num-gpus != 0` when `--data-backend` is equal to  `dali-gpu`. If it happened, the program will generate corresponding warning AND it will set `opt.num_gpus` to the total numbers of GPUs installed on the computer.
- I also see better performance for 8 GPUs. It seems that is because of the updated version of DALI. Later today I will check the performance for other numbers of GPUs and I will update charts and [the log files](https://github.com/andrei5055/gluonCV_logs).